fission-server/migrations/2023-06-15-232702_volumes/up.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE volumes (
+    id SERIAL PRIMARY KEY,
+
+    inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMP NOT NULL DEFAULT NOW(),
+
+    cid TEXT NOT NULL
+);
+
+ALTER TABLE accounts
+  ADD COLUMN volume_id INTEGER REFERENCES volumes(id);
+
+ALTER TABLE apps
+  ADD COLUMN volume_id INTEGER REFERENCES volumes(id);
```

Comment:
If I'm reading this correctly, this means that the volume id for an app should be the same as the app's developer volume id. Right? Because apps are uploaded to the file system of the developer.

Reply:
Oh, good point. I'm not sure what the best thing to do is – I was thinking that apps had their own storage, but we definitely have discussed that they don't. My bias at least to start would be to have a strictly normalized schema, and only denormalize if we find it's a problem, so I'll remove the apps->volume link since we can transitively fetch it via the controlling account.

It'll be an extra two queries, but we'll keep an eye on that and if it turns out it's too expensive we can denormalize. My hunch is we'll get to a pretty significant scale before we notice the extra ~20 ms.